axelrod/tests/strategies/test_zero_determinant.py

```diff
@@ -317,3 +317,46 @@ def test_strategy(self):
 
         actions = [(C, D), (D, C), (D, D), (D, C), (D, D), (D, C)]
         self.versus_test(opponent=axelrod.CyclerDC(), expected_actions=actions, seed=5)
+
+
+class TestAdaptiveZeroDet(TestPlayer):
+    """
+    Test the AdaptiveZeroDet strategy.
+    """
+
+    name = "AdaptiveZeroDet: 0.125, 0.5, 3, C"
+    player = axelrod.AdaptiveZeroDet
+    expected_classifier = {
+        "memory_depth": float('inf'),
+        "stochastic": True,
+        "makes_use_of": set(['game']),
+        "long_run_time": False,
+        "inspects_source": False,
+        "manipulates_source": False,
+        "manipulates_state": False,
+    }
+
+    def test_strategy(self):
+        R, P, S, T = axelrod.Game().RPST()
+
+        player = self.player(l=R)
+        axelrod.seed(0)
+        match = axelrod.Match((player, axelrod.Alternator()), turns=200)
+        match.play()
```

Comment:
This tests need to be tweaked so that we can check expected behaviours. There's information on writing tests here: https://axelrod.readthedocs.io/en/stable/tutorials/contributing/strategy/writing_test_for_the_new_strategy.html

For example this specific paragraph would be something like (although I am not sure what the expected behaviour is):

```suggestion
        expected_actions = [(C,D), (D, C)] * 100 
        self.versus_test(opponent=axelrod.Alternator),
                 expected_actions=expected_actions, seed=0, init_kwargs={"l": R})
```

Note that we could also check that the other attributes have changed accordingly using the `attrs` keyword.

Let me know if I can assist with that (happy to PR the tests to your branch).

Reply:
I agree but I didn't want to search for seeds just to have to do so again once #1288 goes in. The current tests make sure that at least each branch is run (so that the limits are bounced into).

Reply:
I understand. Are you suggesting this are place holders and we merge #1288 and then fix these?

Reply:
Yes, after we merge this I'll update all the tests that require seeds in #1288 (there are many and most will have to change).

Reply:
Sounds good. Happy to take a share of that work if helpful.